### DISCUSSION TITLE
Setting the order of execution

### DISCUSSION THREAD

**Discussion by Mastomaki** [Spine-project/Spine-Toolbox#2516] [Support]
Hello!

How do I set the order of execution for components? Now it looks like this:

![Screenshot](https://github.com/spine-tools/Spine-Toolbox/assets/42839563/13d08309-555c-498d-91b7-4f04d2aaea9e)

Can I also define a group, which involves running a bunch of components in specific order?

**Comment by ptsavol** [comment]
The order of execution is set automatically based on the links you add between project items.

**Reply by Mastomaki** [reply]
How about if there is Importer1, which imports objects, and Importer2 which imports parameters for the objects, then the first one should be run first. There is no direct link between these.

**Reply by ptsavol** [reply]
I think you are looking for the write index that can be set in the link properties.

![discussion_2516](https://github.com/spine-tools/Spine-Toolbox/assets/43600314/363d9e01-8146-46f4-bc42-7575009dfcf5)

**Reply by Mastomaki** [reply]
Ok, that sounds like it could do the job.

**Comment by PiispaH** [comment]
The order of execution should also be visible on the project items in the upper left corner. If you are using system wide dark mode, the number indicating the number might not be visible.

**Reply by Mastomaki** [reply]
As far as I know, I'm not using dark mode.

**Comment by Mastomaki** [comment]
If two importers have the same write index, does the toolbox run them in the right order, e.g. objects before relationships?

**Reply by ptsavol** [reply]
As far as I know, there is no logic implemented to solve these kinds of situations. I.e. the user should make sure the write index is correct in cases like these.

**Reply by manuelma** [reply]
I think in this case you as a user need to make sure each importer is self-contained, that is importer B doesn't depend on work done by importer A and viceversa. But of course each importer imports objects before relationships using those objects.
